lms/djangoapps/instructor/views/api.py

```diff
@@ -2297,50 +2320,77 @@ def list_email_content(request, course_id):  # pylint: disable=unused-argument
     return JsonResponse(response_payload)
 
 
-@require_POST
-@ensure_csrf_cookie
-@cache_control(no_cache=True, no_store=True, must_revalidate=True)
-@require_level('staff')
-def list_instructor_tasks(request, course_id):
+class InstructorTasks(DeveloperErrorViewMixin, APIView):
     """
-    List instructor tasks.
+    **Use Cases**
+
+        Lists currently running instructor tasks
 
-    Takes optional query paremeters.
+    **Parameters**
         - With no arguments, lists running tasks.
-        - `problem_location_str` lists task history for problem
+        - `problem_location_str` lists task history for problemself.
         - `problem_location_str` and `unique_student_identifier` lists task
             history for problem AND student (intersection)
+
+    **Example Requests**:
+
+        POST /courses/{course_id}/instructor/api/list_instructor_tasks
+
+    **Response Values**
+        {
+          "tasks": [
+            {
+              "status": "Incomplete",
+              "task_type": "grade_problems",
+              "task_id": "2519ff31-22d9-4a62-91e2-55495895b355",
+              "created": "2019-01-15T18:00:15.902470+00:00",
+              "task_input": "{}",
+              "duration_sec": "unknown",
+              "task_message": "No status information available",
+              "requester": "staff",
+              "task_state": "PROGRESS"
+            }
+          ]
+        }
     """
-    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
-    problem_location_str = strip_if_string(request.POST.get('problem_location_str', False))
-    student = request.POST.get('unique_student_identifier', None)
-    if student is not None:
-        student = get_student_from_identifier(student)
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, IsCourseStaff)
 
-    if student and not problem_location_str:
-        return HttpResponseBadRequest(
-            "unique_student_identifier must accompany problem_location_str"
-        )
+    @cache_control(no_cache=True, no_store=True, must_revalidate=True)
+    def post(self, request, course_id):
+        """
+        List instructor tasks.
+        """
+        course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
```

Comment:
Could you change instances of SlashSeparatedCourseKey... to CourseKey...
See discussion [here](https://github.com/edx/edx-platform/pull/18975#discussion_r222773392)